gateway_test.go

```diff
@@ -24,9 +24,66 @@ func init() {
 	runningTests = true
 }
 
+// to register to, but never used
 var discardMuxer = mux.NewRouter()
 
+const (
+	// we need a static port so that the urls can be used in static
+	// test data and the requests aren't randomized for checksums
+	// port 16500 should be obscure and unused
+	testHttpListen = "127.0.0.1:16500"
```

Comment:
Read the comment :)

I agree that that would be better, but it would need refactoring so that the tests won't fail. Right now some depend on the url being static.